### ISSUE TITLE
Rule S1172: FN in ctors

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#5011]
### Description

Rule [S1172](https://rules.sonarsource.com/csharp/RSPEC-1172) does not raise for constructor parameters.

### Repro steps

```
using System;

public class A
{
    public A(int i, int j) // FN
    {
        Console.WriteLine(j);
    }
}

public record class B
{
    public B(int i, int j) // FN
    {
        Console.WriteLine(j);
    }
}

public struct C
{
    public C(int i, int j) // FN
    {
        Console.WriteLine(j);
    }
}

public record struct D
{
    public D(int i, int j) // FN
    {
        Console.WriteLine(j);
    }
}

```

### Expected behavior

In all these cases `i` parameter is not used and an issue should be raised.

### Actual behavior

No issues are raised.

### Related information

* C#/VB.NET Plugins version: 8.30
